### PR TITLE
Upgrade packit.yaml config to have integration tests

### PR DESCRIPTION
**Pull request by r0x0d**
This commit introduces the execution of leapp-repository integration tests as a packit job.

### CODE REVIEW

File: .packit.yaml [hunk 1]
```diff
@@ -84,3 +84,173 @@ jobs:
...
+- job: tests
+  fmf_url: "https://gitlab.cee.redhat.com/oamg/tmt-plans"
+  fmf_ref: "master"
+  use_internal_tf: True
+  trigger: pull_request
+  targets:
+    epel-7-x86_64:
+      distros: [RHEL-7.9-ZStream]
+  identifier: tests-7.9to8.6
+  tmt_plan: "^(?!.*c2r)(?!.*sap)(?!.*8to9)"
```
Comment (fernflower) [inline .packit.yaml:97]: Should be `^(?!.*upgrade_plugin)(?!.*tier[2-3].*)(?!.*rhsm)(?!.*c2r)(?!.*sap)(?!.*8to9)(?!.*max_sst)`

You can use https://github.com/oamg/leapp-repository/blob/master/.github/workflows/tmt-tests.yml for reference

File: .packit.yaml [hunk 2]
```diff
@@ -84,3 +84,173 @@ jobs:
...
+
+- job: tests
+  fmf_url: "https://gitlab.cee.redhat.com/oamg/leapp-tests"
+  fmf_ref: "master"
+  use_internal_tf: True
+  trigger: pull_request
+  targets:
+    epel-7-x86_64:
+      distros: [RHEL-7.9-ZStream]
+  identifier: tests-7.9to8.8-sst
```
Comment (fernflower) [inline .packit.yaml:136]: Let's drop -sst tests here completely, it doesn't make sense to run them for every pr. Let's wait till packit adds a feature to run subset of tests on demand.

File: .packit.yaml [hunk 3]
```diff
@@ -84,3 +84,173 @@ jobs:
...
+      distros: [RHEL-7.9-ZStream]
+  identifier: tests-7.9to8.6
+  tmt_plan: "^(?!.*c2r)(?!.*sap)(?!.*8to9)"
+  tf_post_install_script: "#!/bin/sh\nsudo sed -i s/.*ssh-rsa/ssh-rsa/ /root/.ssh/authorized_keys"
+  tf_extra_params:
+    environments:
+      - tmt:
+          context:
+            distro: "rhel-7.9"
+  env:
```
Comment (fernflower) [inline .packit.yaml:104]: Also please add `LEAPPDATA_BRANCH=upstream` as env var for every test, like https://github.com/oamg/leapp-repository/blob/master/.github/workflows/tmt-tests.yml#L34

File: .packit.yaml [hunk 4]
```diff
@@ -84,3 +84,173 @@ jobs:
...
+      - tmt:
+          context:
+            distro: "rhel-8.6"
+  env:
+    SOURCE_RELEASE: "8.6"
+    TARGET_RELEASE: "9.0"
+    TARGET_KERNEL: "el9"
+    RHSM_REPOS: "rhel-8-for-x86_64-appstream-eus-rpms,rhel-8-for-x86_64-baseos-eus-rpms"
+
+- job: tests
```
Comment (fernflower) [inline .packit.yaml:191]: 8.7->9.0 is not a supported upgrade path anymore, please change the test to be 8.8->9.2 as here https://github.com/oamg/leapp-repository/blob/master/.github/workflows/tmt-tests.yml#L86

**Review by fernflower — CHANGES_REQUESTED** [review]
Overall, I have a very good feeling about this, thank you for the contribution!

File: .packit.yaml (hunk 2 above)
Comment (r0x0d) [reply .packit.yaml:136]: I will just comment them out to make it easier to add it back when packit adds the feature you guys need.

File: .packit.yaml (hunk 1 above)
Comment (r0x0d) [reply .packit.yaml:97]: done.

File: .packit.yaml (hunk 2 above)
Comment (r0x0d) [reply .packit.yaml:136]: done.

File: .packit.yaml (hunk 3 above)
Comment (r0x0d) [reply .packit.yaml:104]: done.

**Comment by fernflower** [comment]
Awesome, just 1 test is failed and it will be solved once https://github.com/oamg/leapp-repository/pull/1052 is merged.

**Pull request merged**